turnToHeading()

nic-cheng opened the pull request:
#### Summary
Implements turn to heading function, renamed old function to `turnToPoint()`.

#### Motivation
Turning to a heading instead of a point is also useful.

#### Test Plan
Plan: run on a robot the same way turnToPoint and other motions would be tested.

#### Additional Notes
Runs with slightly modified logic from the old `turnTo()` so there should be no issues (unless I screwed something up).
Tested by @edwart2cAKE.

<!-- DO NOT REMOVE!! -->
<!-- bot: nightly-link -->
<!-- commit-sha: 015114bfb2bdc9b0d23fc704dcd053410c3ceefa -->
## Download the template for this pull request: 

> [!NOTE]  
> This is auto generated from [`Add Template to Pull Request`](https://github.com/LemLib/LemLib/actions/runs/7696942468)
- via manual download: [LemLib@0.5.0-rc.6+015114.zip](https://nightly.link/LemLib/LemLib/actions/artifacts/1202490140.zip)
- via PROS Integrated Terminal: 
 ```
curl -o LemLib@0.5.0-rc.6+015114.zip https://nightly.link/LemLib/LemLib/actions/artifacts/1202490140.zip;
pros c fetch LemLib@0.5.0-rc.6+015114.zip;
pros c apply LemLib@0.5.0-rc.6+015114;
rm LemLib@0.5.0-rc.6+015114.zip;
```

Commit:
Create files + PistonGroup class

Commit:
Merge branch 'master' of https://github.com/Miniongolf/LemLib

Commit:
add pto class constructors and method declaration

Commit:
Implement `turntoHeading()`

Renamed old `turnTo()` to `turnToPoint()`.

Commit:
Remove all references to PTO, update main.cpp

Raptorly1 committed:
turnToHeading() clang-format

nic-cheng committed:
Merge pull request #3 from Raptorly1/turnToHeading()-clang-format

turnToHeading() clang-format

liam-teale commented on include/lemlib/chassis/chassis.hpp:
```diff
@@ -236,7 +236,22 @@ class Chassis {
+         * @param maxSpeed the maximum speed the robot can turn at. Default is 127
+         * @param async whether the function should be run asynchronously. true by default
+         */
+        void turnToHeading(float targetTheta, int timeout, bool radians = false, bool forwards = true,
```
there should not be an option for radians. It's highly unlikely a user will use radians, and it makes the function call more complex

liam-teale commented on src/lemlib/chassis/chassis.cpp:
```diff
@@ -279,6 +279,78 @@ void lemlib::Chassis::turnTo(float x, float y, int timeout, bool forwards, float
+        motorPower = angularPID.update(deltaTheta);
+        angularLargeExit.update(deltaTheta);
+        angularSmallExit.update(deltaTheta);
+
```
The output should be slewed unless its within 20 degrees of the target

liam-teale commented on src/main.cpp:
```diff
@@ -131,9 +132,13 @@ void autonomous() {
+    // example movement: Turn to face a heading of 90º. Timeout set to 1500
+    chassis.turnToHeading(1000, 1500);
+    // example movement: Turn to face a heading of pi radians. Timeout set to 1500
+    chassis.turnToHeading(M_PI, 1500, true);
```
remember to remove this when you remove the radians option

Commit:
Fix according to @SizzinSeal's review

Slew code will be updated to match turnToPoint() once the updated version of turnToPoint() is pushed.

Commit:
Fix clang-format

liam-teale reviewed (approved):
LGTM, will merge upon testing

Commit:
Fix main.cpp example heading

liam-teale commented on .DS_Store:
This should not be here. Remove it, and I'll also add it to the gitignore

liam-teale commented on include/lemlib/api.hpp:
```diff
@@ -10,10 +10,11 @@
 #include "lemlib/pose.hpp"
-#include "lemlib/chassis/trackingWheel.hpp"
+#include "lemlib/util.hpp"
+
```
why the newlines in this file?

liam-teale committed:
Merge branch 'master' into turnToHeadingPR

Commit:
Remove random `api.hpp` newlines

Commit:
Update .gitignore

Commit:
Merge remote-tracking branch 'upstream/master' into turnToHeadingPR

Commit:
Fix pr by re-introducing params (hopefully)

Commit:
Update main.cpp

Commit:
Add overloads without params

Commit:
Fix clang-format
